src/simulate.rs

```diff
@@ -91,19 +93,74 @@ pub fn simulate(
     warn!("\tPlayer 0: {:?}({})", player_codes[0], deck_a_path);
     warn!("\tPlayer 1: {:?}({})", player_codes[1], deck_b_path);
 
+    if let Some(export_path) = export_path {
+        run_simulation_with_export(deck_a_path, deck_b_path, player_codes, num_simulations, seed, export_path);
+    } else {
+        let stats_collector = Box::new(StatsCollector::new());
+        let mut composite_handler = CompositeSimulationEventHandler::new();
+        composite_handler.add_handler(stats_collector);
+        let mut simulation = Simulation::new(
+            deck_a_path,
+            deck_b_path,
+            player_codes,
+            num_simulations,
+            seed,
+            composite_handler,
+        )
+        .expect("Failed to create simulation");
+        simulation.run();
+    }
```

Comment:
Could we make this code look like this instead?

```rust
let stats_collector = Box::new(StatsCollector::new());
let mut composite_handler = CompositeSimulationEventHandler::new();
composite_handler.add_handler(stats_collector);
if (user_wants_export) {
  composite_handler.add_handler(training_handler);
}
```

Reply:
That way, we also save on the `run_simulation_with_export` which duplicates a bit the main game loop as well (we would have to maintain that secondary game loop otherwise).